docs/README.md

```diff
@@ -0,0 +1,40 @@
+# COMPOSITE BUILDS
+
+## Challenge
+
+When using a model builder to fetch the GradleSourceSet model, the complete hierarchical scope of the project is lost.
+This hinders the ability to fetch included builds and project dependencies, which is crucial for composite builds.
+
+## Our Solution: Using Build Actions
+
+To address this limitation we take the following approach:
+
+1. We utilize the `BuildAction` to capture the complete hierarchical scope of the project, allowing us to retrieve all
+   the included builds and their associated information.
+2. After obtaining the included builds, we use the model builder to get source sets for each individual project.
+3. With te complete set of source sets for each included build we map the build target dependencies and return the list
+   of `GradleSourceSets`.
+
+![Composite Build](../images/composite_build.png)
+
+# JAVA HOME HANDLING
```

Comment:
Please do not use Upper Case. If the documentation engine needs it, they can apply it via CSS style.

I also recommend using first level (`#`) as a unique title in the document top, and use lower levels for sections

Reply:
Addressed in https://github.com/microsoft/build-server-for-gradle/pull/189/commits/b624852d90f8acc723f4bb12128d95648ac18442